config/config_test.go

```diff
@@ -541,6 +550,106 @@ func TestGetSubnetConfigsFromFlags(t *testing.T) {
 	}
 }
 
+type signerServer struct {
+	signer.UnimplementedSignerServer
+}
+
+func (*signerServer) PublicKey(context.Context, *signer.PublicKeyRequest) (*signer.PublicKeyResponse, error) {
+	// for tests to pass, this must be the base64 encoding of a 32 byte public key
+	// but it does not need to be associated with any private key
+	bytes, err := base64.StdEncoding.DecodeString("j8Ndzc1I6EYWYUWAdhcwpQ1I2xX/i4fdwgJIaxbHlf9yQKMT0jlReiiLYsydgaS1")
+	if err != nil {
+		return nil, err
+	}
+
+	return &signer.PublicKeyResponse{
+		PublicKey: bytes,
+	}, nil
+}
+
+func TestGetStakingSigner(t *testing.T) {
+	testKey := "HLimS3vRibTMk9lZD4b+Z+GLuSBShvgbsu0WTLt2Kd4="
+	rpcServer := grpc.NewServer()
+	defer rpcServer.GracefulStop()
+
+	signer.RegisterSignerServer(rpcServer, &signerServer{})
+
+	listener, err := net.Listen("tcp", "[::1]:0")
+	require.NoError(t, err)
+
+	go func() {
+		err := rpcServer.Serve(listener)
+		require.NoError(t, err)
+	}()
+
+	type config map[string]any
+	type test map[string]struct {
+		viperKeys          string
+		config             config
+		expectedSignerType reflect.Type
+		expectedErr        error
+	}
```

Comment:
nit: you can define these anonymously like so (and also avoid the map-style initialization that is inconsistent w/ the way we have most of our tests written)
````
	tests := []struct{
		name string
		foo     bar
	} {
		{
			name: "foo",
			foo:  bar{},
		},
	}

	for _, tt := range tests {
      // test here
	}
````